romancal/source_catalog/injection.py

```diff
@@ -40,3 +69,120 @@ def inject_sources(model, si_cat):
+    filter : str
+        Name of filter (used to calculate size)
+    seed : int
+        Seed for random number generator
+
+    Returns
+    -------
+    all_cat : astropy.Table
+        Table for use with table_to_catalog to generate catalog for simulation.
+    """
+    from romanisim import bandpass, catalog
+
+    # WFI bandpasses
+    BANDPASSES = set(bandpass.galsim2roman_bandpass.values())
+
+    # Set random source index for the catalog
+    rng_numpy = np.random.default_rng(seed)
+    ran_idx = rng_numpy.permutation(len(xpos))
+
+    # 75% of objects as galaxies, 25% of objects as point sources
+    num_stars = int(len(xpos) / 4)
+    num_gals = len(xpos) - num_stars
+
+    # Galaxies
+    # RomanISim's make cosmos galaxies method will return cosmos like objects with
+    # uniformly distributed position angles. Radius and area are set to return
+    # a full sample (~346k galaxies)
+    gal_cat = catalog.make_cosmos_galaxies(cen, radius=1.0, cat_area=(np.pi), **kwargs)
+
+    # Trim to the required number of objects
+    gal_cat = gal_cat[:num_gals]
+
+    # Set magnitude spread
+    mags = rng_numpy.uniform(low=-4.0, high=1.0, size=num_gals)
+
+    # Brightest color flux
+    gal_cat_data = (
+        gal_cat.as_array(names=BANDPASSES)
+        .view(dtype=float)
+        .reshape((len(gal_cat), len(BANDPASSES)))
+    )
+    deepflux = gal_cat_data.max(axis=1)
+
+    # Set bandpass fluxes
+    for bp in BANDPASSES:
+        # Normalize the mag limit to exptime
+        gal_mag_limit = HRGALMAGLIMIT[bp] + (
+            1.25 * np.log10((exptime * u.s).to(u.hour).value)
+        )
+        mag_tot = mags + gal_mag_limit
+
+        # Set scaled fluxes
+        gal_cat[bp] = (gal_cat[bp] * (10.0 ** (-(mag_tot) / 2.5)) / deepflux).astype(
+            "f4"
+        )
+
+    # Sizes are drawn from a log-normal distribution of J-band magnitudes
+    # The parameters below are derived from the distribution for sizes
+    mu = (-0.1555 * ((-2.5 * np.log10(gal_cat["F129"])) - 17)) - 3.55
+    sigma = 0.15
+    radsize = rng_numpy.normal(mu, sigma)
+    gal_cat["half_light_radius"] = (10**radsize) * 3600 * u.arcsec
+
+    # Randomize concentrations
+    gal_cat["n"] = rng_numpy.uniform(low=0.8, high=4.5, size=num_gals)
+
+    # Stars
+    # Create base table
+    star_cat = table.Table()
+    star_cat["ra"] = num_stars * [0]
+    star_cat["dec"] = num_stars * [0]
+    star_cat["type"] = num_stars * ["PSF"]
+    star_cat["n"] = num_stars * [-1]
+    star_cat["half_light_radius"] = num_stars * [0]
+    star_cat["pa"] = num_stars * [0]
+    star_cat["ba"] = num_stars * [1]
+
+    # Set magnitude spread
+    mags = rng_numpy.uniform(low=-4.0, high=1.0, size=num_stars)
+
+    # Set bandpass fluxes
+    for bp in BANDPASSES:
+        point_mag_limit = HRPOINTMAGLIMIT[bp] + (
+            1.25 * np.log10((exptime * u.s).to(u.hour).value)
+        )
+        star_cat[bp] = (10.0 ** (-(mags + point_mag_limit) / 2.5)).astype("f4")
```

Comment:
I think we want to have the stars all have color = 0.  This makes all of the stars have colors point_mag_limit[x] - point_mag_limit[y]?

Reply:
It does, which is what I thought you wanted it changed to in your previous comment (when switching to per band point mag limit). I can add an offset to preserve color = 0.